Add utility to docs to convert YAML config to env vars and back

jacobtomlinson opened the pull request:
A few times lately I've had folks ask me how to set Dask configuration via environment variables. The docs have a section on this, but most of the config examples throughout the rest of the documentation are in YAML or in the dot notation. The task of converting between them is confusing to new users.

This PR adds a small utility to the config documentation which allows users to convert between YAML, environment variable and Python dot notation configuration. My hope here is that people will discover it when looking for how to set config via environment variables and then use it as a copy/paste tool to convert other config they are trying to set.

The JavaScript included in this PR is a reproduction of some of the config code to enable the conversion between the three forms. Due to the rewrite and differences between Python and JavaScript it probably isn't a perfect implementation, but I'm confident that it is similar enough to be useful.

![Demo](https://user-images.githubusercontent.com/1610850/88934275-590aa600-d278-11ea-9f93-2dd55c57f0fa.gif)

quasiben commented:
Oh that's really nice!  I have also been flummoxed by this in the past.  Do you think we should also include a reference to this on the https://docs.dask.org/en/latest/configuration-reference.html page ?

jacobtomlinson commented:
Good point! That page also uses the dot notation which is common to use within Python code. I could add a third box which shows some Python code for setting on the fly too.

```python
>>> with dask.config.set({'foo.bar': 123}):
...     pass
```

jacobtomlinson commented:
Updated config reference page as suggested by @quasiben and added dot notation support.

![image](https://user-images.githubusercontent.com/1610850/88941796-5a8c9c00-d281-11ea-8c3b-51aac55e5a50.png)

jrbourbeau commented on docs/source/configuration.rst:
```diff
@@ -152,6 +152,48 @@ as lists, dictionaries, and so on with normal Python syntax.
+Conversion Utility
+^^^^^^^^^^^^^^^^^^
+
+Many examples in the documentation use YAML config in their examples. You can enter
```
What do you think about using the text you added over in the configuration reference here? It seems a little more clear to me that there are several ways to set config values and this utility helps you convert between them. 

```suggestion
It is possible to configure Dask inline with dot notation, with YAML or via environment variables. You can enter
```

jrbourbeau commented on docs/source/configuration.rst:
```diff
@@ -152,6 +152,48 @@ as lists, dictionaries, and so on with normal Python syntax.
 Environment variables take precedence over configuration values found in YAML
 files.
 
+Conversion Utility
```
Nitpick: This is a subsection under the "Environment Variables" section, but it also makes use of YAML and `dask.config.set`. What do you think about promoting this section to be at the same level as "Environment Variables", "YAML files", and "Directly within Python?

jacobtomlinson replied:
Good idea! This was originally for YAML -> env vars. But it has since expanded.

jacobtomlinson replied:
I have promoted the heading and also repositioned it at the end of the current section.

jakirkham commented:
I don't know how I didn't come across this until now, but this is absolutely amazing! Thank you for adding this Jacob 😄
